js/bundle_test.go

```diff
@@ -675,3 +696,85 @@ func TestBundleEnv(t *testing.T) {
 		})
 	}
 }
+
+func TestBundleMakeArchive(t *testing.T) {
+	t.Run("ok", func(t *testing.T) {
+		testCases := []struct {
+			cm      compiler.CompatibilityMode
+			script  string
+			exclaim string
+		}{
+			{compiler.CompatibilityModeExtended,
+				`import exclaim from "./exclaim.js";
+				export let options = { vus: 12345 };
+				export let file = open("./file.txt");
+				export default function() { return exclaim(file); };`,
+				`export default function(s) { return s + "!" };`},
+			{compiler.CompatibilityModeBase,
+				`var exclaim = require("./exclaim.js");
+				module.exports.options = { vus: 12345 };
+				module.exports.file = open("./file.txt");
+				module.exports.default = function() { return exclaim(module.exports.file); };`,
+				`module.exports.default = function(s) { return s + "!" };`},
```

Comment:
nit: again the code looks wonky at least in  github

Reply:
This looks a bit better than the other case, no? :)

The misalignment on lines 712 and 718 is because that's a new string, so that should be fine. I'll fix the ones on 708 and 714.

Reply:
Fixed in 45407364.